hr_expense_advance_clearing/models/hr_expense.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2019 Kitti Upariphutthiphong <kittiu@ecosoft.co.th>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class HrExpense(models.Model):
+    _inherit = 'hr.expense'
+
+    advance = fields.Boolean(
+        string='Employee Advance',
+        default=False,
+    )
+
+    @api.multi
+    @api.constrains('advance')
+    def _check_advance(self):
+        emp_advance = self.env.ref('hr_expense_advance_clearing.'
+                                   'product_emp_advance')
+        for expense in self.filtered('advance'):
+            if not emp_advance.property_account_expense_id:
```

Comment:
This if must be moved out of the for, you are not using the expense variable here

Reply:
Yes! Done it, thank you!